yt_dlp/utils.py

```diff
@@ -840,11 +840,17 @@ class Popen(subprocess.Popen):
     else:
         _startupinfo = None
 
-    def __init__(self, *args, text=False, **kwargs):
+    def __init__(self, *args, env=None, text=False, **kwargs):
         if text is True:
             kwargs['universal_newlines'] = True  # For 3.6 compatibility
             kwargs.setdefault('encoding', 'utf-8')
             kwargs.setdefault('errors', 'replace')
+        if sys.platform not in ('win32', 'cygwin'):
```

Comment:
Isn't this better to do at initialization? similar to `windows_enable_vt_mode`

Also, since we only want this to happen for pyinstaller exes, better to check `compat_os_name != 'nt' && hasattr(sys, '_MEIPASS')`

Reply:
I think mutating the environ also changes the environment variables for the running process...
Agreed for changing decision logic

Reply:
> I think mutating os.environ also changes the environment variables for the running process...

and is that a problem?

Reply:
Unsetting it on runtime would make it load the undesired shared objects that PyInstaller didn't expect (i.e. load ones on system but bundled one by PI)